pyro/infer/mcmc/util.py

```diff
@@ -384,6 +384,8 @@ def initialize_model(model, model_args=(), model_kwargs={}, transforms=None, max
     prototype_samples = {}
     for name, node in model_trace.iter_stochastic_nodes():
         if isinstance(node["fn"], _Subsample):
+            if node["fn"].subsample_size:
```

Comment:
I think we can be more permissive and check:
```py
fn = node["fn"]
if fn.subsample_size is not None or fn.subsample_size < fn.size:
    raise ...
```
This allows passing `subsample_size=len(data)` for models that support either MCMC or SVI.